Build a regression model.

In [1]:
import pandas as pd
import statsmodels.api as sm

In [2]:
df = pd.read_csv("df_yelp_full.csv")


In [4]:
df = df.dropna(subset=["free_bikes", "rating"])  

X = df[["rating"]]
X = sm.add_constant(X)
y = df["free_bikes"]

In [5]:
model = sm.OLS(y, X).fit()

Provide model output and an interpretation of the results. 

In [6]:
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:             free_bikes   R-squared:                       0.027
Model:                            OLS   Adj. R-squared:                  0.026
Method:                 Least Squares   F-statistic:                     19.30
Date:                Tue, 08 Jul 2025   Prob (F-statistic):           1.29e-05
Time:                        20:03:22   Log-Likelihood:                -1584.1
No. Observations:                 698   AIC:                             3172.
Df Residuals:                     696   BIC:                             3181.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.2126      0.117     18.918      0.0

The regression model examined at whether the number of free bikes at a station could be explained by the Yelp ratings of nearby places like restaurants and bars. The result showed a small positive relationship: stations near higher-rated spots had slightly more bikes available. For each extra star in rating, the number of bikes went up by about 0.18. But the R-squared was only 0.027, which means the model doesn’t explain much of what’s really going on. So even though the rating has some influence, it’s pretty limited, other factors are probably playing a bigger role.

# Stretch

How can you turn the regression model into a classification model?

To turn this into a classification model, I’d group the number of bikes into categories like “low” (0–2 bikes), “medium” (3–5), and “high” (6 or more). Instead of predicting the exact number, the model would try to predict which group a station falls into. Then I’d use something like logistic regression to see if things like the number of nearby restaurants, their type, or their ratings affect the chance of a station being in one category or another. This would help simplify the output and maybe even give more useful insights.

